plugins/package-managers/node/src/main/kotlin/Npm.kt

```diff
@@ -534,8 +534,13 @@ open class Npm(
 
         val rawName = json["name"].textValueOrEmpty()
         val (namespace, name) = splitNpmNamespaceAndName(rawName)
-        if (name.isBlank()) {
-            logger.warn { "'$packageJson' does not define a name." }
+
+        val projectName = if (name.isBlank()) {
+            getFallbackProjectName(analysisRoot, packageJson).also {
+                logger.warn { "'$packageJson' does not define a name, falling back to '$it'." }
```

Comment:
We have a couple of callers of `getFallbackProjectName()`, but none logs any message about the use of the fallback. What doy ou think about simply dropping this here as well?

Reply:
Then, code can be shortened, e.g. to:
`= name.takeUnless { it.isBlank() } ?: getFallbackProjectName(analysisRoot, packageJson)`

Reply:
> What doy ou think about simply dropping this here as well?

This is basically just the previously existing log statement, extended with the fallback name. It did not feel right to me to remove that log statement altogether in this context, esp. as there is a similar log statement about the version, too, which should then also be removed for consistency.

Reply:
However, I've simplified the code via `ifBlank {}` now.